stix/idl/processing/pixel_data/stx_construct_pixel_data.pro

```diff
@@ -0,0 +1,393 @@
+;     - DETECTOR_MASKS: matrix containing information on the detectors used in the selected time interval
+;
+; KEYWORDS:
+;
+;   path_bkg_file: path of a background L1 fits file. If provided, the fields 'COUNTS_BKG', 'COUNTS_ERROR_BKG' and 'LIVE_TIME_BKG' 
+;                  of the pixel_data structure are filled with the values read from the background measurement file
+;
+;   elut_corr: if set, a correction based on a ELUT table is applied to the measured counts
+;   
+;   xy_flare: two-element array containing the X and Y coordinates of the estimated flare location
+;             (STIX coordinate frame, arcsec). If passed, the grid transmission correction is computed
+;   
+;   subc_index: array containing the indices of the selected imaging detectors. Used only for plotting the lightcurve by means of 
+;             'stx_plot_selected_time_range'. Default, indices of
+;             the detectors from 10 to 3
+;   
+;   sumcase: string indicating which pixels are considered. Used only for plotting the lightcurve by means of
+;            'stx_plot_selected_time_range'. Refer to the header of 'stx_sum_pixel_data' for more details.
+;             Default, 'TOP+BOT'
+;    
+;   silent: if set, plots are not displayed
+;   
+;   no_small: if set, Moire patterns measured by small pixels are not plotted with 'stx_plot_moire_pattern'
+;   
+;   no_rcr_check: if set, control on RCR change during the selected time interval is not performed. Default, 0
+;   
+;   shift_duration: if set, shift all time bins by 1 to account for FSW time input discrepancy prior to 09-Dec-2021.
+;                   N.B. WILL ONLY WORK WITH FULL TIME RESOLUTION DATA WHICH IS OFTEN NOT THE CASE FOR PIXEL DATA.
+;             
+; HISTORY: July 2022, Massa P., created
+;          September 2022, Massa P., added 'shift_duration' keyword
+;
+; CONTACT:
+;   paolo.massa@wku.edu
+;-
+
+function stx_construct_pixel_data, path_sci_file, time_range, energy_range, elut_corr=elut_corr, $
+                                   path_bkg_file=path_bkg_file, xy_flare=xy_flare, subc_index=subc_index, $
+                                   sumcase=sumcase, silent=silent, no_small=no_small, no_rcr_check=no_rcr_check, $
+                                   shift_duration=shift_duration, _extra=extra
+
+default, elut_corr, 1
+default, silent, 0
+default, subc_index, stx_label2ind(['10a','10b','10c','9a','9b','9c','8a','8b','8c','7a','7b','7c',$
+                                   '6a','6b','6c','5a','5b','5c','4a','4b','4c','3a','3b','3c'])
+
+default, sumcase, "TOP+BOT"
+default, no_rcr_check, 0
+
+if anytim(time_range[0]) gt anytim(time_range[1]) then message, "Start time is greater than end time"
+if energy_range[0] gt energy_range[1] then message, "Energy range lower edge is greater than the higher edge"
+
+
+stx_read_pixel_data_fits_file, path_sci_file, data_str = data, t_axis = t_axis, e_axis = e_axis, alpha = alpha, shift_duration=shift_duration, _extra=extra
```

Comment:
what `alpha` L1A v L1?

Reply:
Yes, the alpha keyword means the file in question is L1A rather than L1